### PR TITLE
Oop practice

### PR DESCRIPTION
**Pull request by Vicctorri**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "license": "MIT",
     "platform-check": false,
     "require": {
-        "php": "^7.3|^8.0",
```
Comment (egorbwork) [inline composer.json:9]: please use php 7.4 in this case. As it was identified on one of the last lesson, your image for docker is php 7.4

File: app/Services/DebugRequestActivityLogger.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+namespace App\Services;
+
+use Illuminate\Http\Request;
+
+class DebugRequestActivityLogger extends AbstractRequestActivityLogger
+{
+    protected function collectRequestData(Request $request): array
+    {
+        return [
+            'ipAddress'=> $request->ip(),
```
Comment (egorbwork) [inline app/Services/DebugRequestActivityLogger.php:11]: something familiar.

File: app/Http/Controllers/ArticleController.php [hunk 3]
```diff
@@ -4,12 +4,21 @@
 
 use Illuminate\Http\Request;
 use App\Models\Article;
+use Psr\Log\LoggerInterface;
 
 class ArticleController extends Controller
 {
-    public function show($articleId){
-        
+    public function show($articleId, Request $request, LoggerInterface $logger){
```
Comment (egorbwork) [inline app/Http/Controllers/ArticleController.php]: This code is incomplete and was finished at lesson.

**Review by egorbwork — CHANGES_REQUESTED** [review]
Very late and not finished code.

**Comment by egorbwork** [comment]
Plus is missing service/product logging for new logger, so it is seriosly unfinished.

File: composer.json (hunk 1 above)
Comment (Vicctorri) [reply composer.json:9]: Ok

File: app/Http/Controllers/ArticleController.php (hunk 3 above)
Comment (Vicctorri) [reply app/Http/Controllers/ArticleController.php]: I added missing part and complete this code

Comment (egorbwork) [reply app/Http/Controllers/ArticleController.php]: Not valid changes, re-watch the third lesson.

**Review by egorbwork — CHANGES_REQUESTED** [review]
Solution not fixed.

File: app/Http/Controllers/ArticleController.php (hunk 3 above)
Comment (Vicctorri) [reply app/Http/Controllers/ArticleController.php]: Changed

**Review by egorbwork — CHANGES_REQUESTED** [review]
Still Product/Service logging by ModelLogger is missing.

**Review by egorbwork — APPROVED** [review]
Let it be.

**Pull request merged**